creator-node/src/monitors/MonitoringQueue.js

```diff
@@ -0,0 +1,111 @@
+const Bull = require('bull')
+const redis = require('../redis')
+const config = require('../config')
+const { logger } = require('../logging')
+const { getDatabaseSize } = require('./database')
+
+const MONITORING_REDIS_PREFIX = 'monitoring'
+const PROCESS_NAMES = Object.freeze({
+  monitor: 'monitor'
+})
+
+/**
+ * List of all monitors to run, containing:
+ *  @param {string} name A unique name (for caching in redis)
+ *  @param {number} ttl TTL in seconds for how long a cached value is good for
+ *  @param {function} func The actual work to compute a value. The return value is what is cached.
+ */
+const MONITORS = [
+  {
+    name: 'databaseSize',
+    ttl: 120,
+    func: getDatabaseSize
+  }
+  // TODO: Add more monitors
+]
+
+/**
+ * A persistent cron-style queue that periodically monitors various
+ * health metrics and caches values in redis.
+ *
+ * The queue runs every minute on cron, but individual monitors establish
+ * their own freshness/refresh rate to operate on.
+ *  1. The queue spins up and for each monitor checks to see if it needs a refresh
+ *  2. Refreshes the value and stores the update in redis
+ */
+class MonitoringQueue {
+  constructor () {
+    this.queue = new Bull(
+      'monitoring-queue',
```

Comment:
probably want to randomize this queue name

Reply:
how come? it's a constant value?

Reply:
i found that when restarting the server repeatedly having a stale queue name (even if .empty was called) resulting in weird behavior - not a blocker though

Reply:
i know joe's seeing some issues related to this with notifs, but i think the .empty() covers our bases and have tried restarting, etc. many times locally. if we do observe some behavior on prod, we can change this, but since it's not critical path im not too too concerned